packages/webdriverio/src/scripts/resq.js

```diff
@@ -3,35 +3,25 @@ export const waitToLoadReact = function waitToLoadReact () {
 }
 
 export const react$ = function react$ (selector, props, state) {
-    let elems = window.resq.resq$$(selector)
+    let elem = window.resq.resq$(selector)
         .byProps(props)
         .byState(state)
 
-    if (elems.length === 0) {
+    if (!elem.name) {
         return { message: `React element with selector "${selector}" wasn't found` }
     }
 
-    let elem = elems[0]
-
-    if (elem.node) {
-        return elem.node
-    }
-
-    elem = elem.node || elem.children[0].node
-    return elem
+    return elem.node || elem.children[0].node
 }
 
 export const react$$ = function react$$ (selector, props, state) {
     let elems = window.resq.resq$$(selector)
         .byProps(props)
         .byState(state)
 
-    return [...elems].map((elem) => {
-        if (elem.node) {
-            return elem.node
-        }
+    if (!elems.length) {
+        return { message: `React elements with selector "${selector}" wasn't found` }
+    }
 
-        elem = elem.node || elem.children[0].node
-        return elem
-    })
+    return [...elems].map(elem => elem.node || elem.children[0].node)
```

Comment:
Should be here check `!element.name`?

Reply:
The reason I check for `element.name` in the `react$` function is because the resq library returns an object with at least one private property, `_nodes`. So I assume that if the object doesn't have a property `name` then library didn't find a component.

In contrast, for `react$$` the library will return an empty object if no components have been found. That's why it's safe here to assume that there is at least one component in the array since `elems.length` is true.

Hopefully this makes sense.